pallets/uniques/src/lib.rs

```diff
@@ -0,0 +1,849 @@
+/*
+ * This file is part of the Nodle Chain distributed at https://github.com/NodleCode/chain
+ * Copyright (C) 2020-2022  Nodle International
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#![cfg_attr(not(feature = "std"), no_std)]
+
+//! Handle the ability to notify other pallets that they should stop all
+
+use codec::{Decode, Encode};
+use frame_support::pallet_prelude::MaxEncodedLen;
+use frame_support::traits::{tokens::Balance, Currency};
+use scale_info::TypeInfo;
+use sp_runtime::{
+	traits::{StaticLookup, Zero},
+	RuntimeDebug,
+};
+
+pub use pallet::*;
+pub use weights::WeightInfo as NodleWeightInfo;
+
+pub mod weights;
+
+mod benchmarking;
+#[cfg(test)]
+mod tests;
+
+type AccountIdLookupOf<T> = <<T as frame_system::Config>::Lookup as StaticLookup>::Source;
+pub type BalanceOf<T, I = ()> =
+	<<T as pallet_uniques::Config<I>>::Currency as Currency<<T as frame_system::Config>::AccountId>>::Balance;
+
+#[derive(Clone, Encode, Decode, Eq, PartialEq, RuntimeDebug, Default, TypeInfo, MaxEncodedLen)]
+pub struct ExtraDepositDetails<T: Balance> {
+	/// The cap for the total extra deposit for the collection
+	///
+	/// `total` should never go higher than `limit`.
+	limit: T,
```

Comment:
Any reason for doing it that way VS a maximum deposit on a _per item_ basis?

Reply:
@ETeissonniere the number of items are not limited. So it wouldn't give control to the collection owner over how much in total they are allowing the issuer to put on the minted items.

Reply:
Very correct